File: drug-discovery-ui/backend/app/api/models/admet.py
from typing import Optional, Dict, Any
from pydantic import BaseModel


class AdmetRequest(BaseModel):
    molecule: str
    smiles: Optional[str] = None


class AdmetResponse(BaseModel):
    ok: bool
    absorption: Optional[Dict[str, Any]] = None
    distribution: Optional[Dict[str, Any]] = None
    metabolism: Optional[Dict[str, Any]] = None
    excretion: Optional[Dict[str, Any]] = None
    toxicity: Optional[Dict[str, Any]] = None
    overallAssessment: Optional[str] = None
    regulatoryOutlook: Optional[str] = None
    error: Optional[str] = None

File: drug-discovery-ui/src/services/admet.ts
export type AdmetResponse = {
  ok: boolean
  absorption?: any
  distribution?: any
  metabolism?: any
  excretion?: any
  toxicity?: any
  overallAssessment?: string
  regulatoryOutlook?: string
  error?: string
}

export async function analyzeAdmet(input: { molecule: string; smiles?: string }): Promise<AdmetResponse> {
  const resp = await fetch('/api/v1/admet/analyze', {
    method: 'POST',
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify(input),
  })
  if (!resp.ok) throw new Error(`HTTP ${resp.status}`)
  return (await resp.json()) as AdmetResponse
}
